File: app/src/main/java/com/example/msdc/api/TVDetails.java
package com.example.msdc.api;

import com.google.gson.annotations.SerializedName;

import org.json.JSONObject;

public class TVDetails {
    @SerializedName("name")
    String name;

    @SerializedName("poster_path")
    String posterPath;

    @SerializedName("backdrop_path")
    String backdropPath;

    @SerializedName("episode_run_time")
    int[] episodeRuntime;

    @SerializedName("first_air_date")
    String firstAirDate;

    @SerializedName("last_air_date")
    String lastAirDate;

    @SerializedName("number_of_episodes")
    int numberOfEpisodes;

    @SerializedName("number_of_seasons")
    int numberOfSeasons;

    @SerializedName("original_language")
    String originalLanguage;

    @SerializedName("overview")
    String overview;

    @SerializedName("popularity")
    String popularity;

    @SerializedName("status")
    String status;

    @SerializedName("tagline")
    String tagline;

    @SerializedName("vote_average")
    String voteAverage;

    @SerializedName("vote_count")
    String voteCount;

    @SerializedName("homepage")
    String homepage;

    public String getName() {
        return name;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public int[] getEpisodeRuntime() {
        return episodeRuntime;
    }

    public String getFirstAirDate() {
        return firstAirDate;
    }

    public String getLastAirDate() {
        return lastAirDate;
    }

    public int getNumberOfEpisodes() {
        return numberOfEpisodes;
    }

    public int getNumberOfSeasons() {
        return numberOfSeasons;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public String getOverview() {
        return overview;
    }

    public String getPopularity() {
        return popularity;
    }

    public String getStatus() {
        return status;
    }

    public String getTagline() {
        return tagline;
    }

    public String getVoteAverage() {
        return voteAverage;
    }

    public String getVoteCount() {
        return voteCount;
    }

    public String getHomepage() {
        return homepage;
    }
}

File: app/src/main/java/com/example/msdc/api/TVDetails.kt
package com.example.msdc.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class TVDetails (
    @field:SerializedName("name") val name: String? = null,
    @field:SerializedName("poster_path") val posterPath: String? = null,
    @field:SerializedName("backdrop_path") val backdropPath: String? = null,
    @field:SerializedName("episode_run_time") val episodeRuntime: IntArray,
    @field:SerializedName("first_air_date") val firstAirDate: String? = null,
    @field:SerializedName("last_air_date") val lastAirDate: String? = null,
    @field:SerializedName("number_of_episodes") val numberOfEpisodes: Int? = 0,
    @field:SerializedName("number_of_seasons") val numberOfSeasons: Int? = 0,
    @field:SerializedName("original_language") val originalLanguage: String? = null,
    @field:SerializedName("overview") val overview: String? = null,
    @field:SerializedName("popularity") val popularity: String? = null,
    @field:SerializedName("status") val status: String? = null,
    @field:SerializedName("tagline") val tagline: String? = null,
    @field:SerializedName("vote_average") val voteAverage: String? = null,
    @field:SerializedName("vote_count") val voteCount: String? = null,
    @field:SerializedName("homepage") val homepage: String? = null,
    @field:SerializedName("genres") val genres: ArrayList<GenreResult>? = null
) : Parcelable